### ISSUE TITLE
support passing options to the resolver

### ISSUE THREAD

**Issue by matthewmueller** [makojs/css#5]
https://github.com/makojs/css/blob/fab39aa1dbb2f11364197d763d64b758b16818a8/index.js#L69-L73

Useful for treating other extensions (like .styl) as another default style as well as adding `paths` support for `NODE_PATH`.

Used here: https://github.com/lapwinglabs/www-bundle/blob/164f587f382d08cfcd38fd021a4c85f4f1d80241/index.js#L104-L105

**Comment by dominicbarnes** [comment]
Ah yeah, I meant to allow the `extensions` option to be customizable. I can fix that now.

**Comment by dominicbarnes** [comment]
You can now add `extensions` to the configuration of this plugin. (`.css` is always implied) I don't think there's any need for configuring other aspects of `resolve()`, but feel free to re-open if you think that's the case.

**Comment by matthewmueller** [comment]
awesome, thanks! personally, I heavily use `NODE_PATH`s. It's really helpful for quickly open sourcing local modules. So that would require an update to the `paths: [...]` option. It could be done internally though `if (process.env.NODE_PATH) { ... }`.

Given #6, I need to see if NODE_PATHs would actually work with relative assets.

**Comment by dominicbarnes** [comment]
I'll think about adding `options.resolveParams` which will extend the params being passed to `resolve()` (I'll do the same in JS as well) However, I think it would be far easier to just use `npm link`, or using `npm install user/repo`. (would that work?)

As far as the assets, I am thinking that files that are "above" the root could be problematic, as their structure in the output directory is a bit ambiguous, which is why the above 2 options seem best to me.

**Comment by matthewmueller** [comment]
> However, I think it would be far easier to just use npm link, or using npm install user/repo. (would that work?)

This is sort of the other way, when you develop something in your project and are ready to open source it, you don't have to change all the references. It also just cleans things up:

```css
@import '../../lib/font-raleway/index.css';
```

becomes

```css
@import 'font-raleway/index.css';
```

I'm down with whatever you decide, but I would really like to see this option available in some form or another.

**Comment by dominicbarnes** [comment]
If that is what you're wanting to do, then `npm link` is byfar the best way to do it, anything else seems to introduce way more complexity especially considering the simpler alternative.

**Comment by matthewmueller** [comment]
Hmm, maybe I'm not understanding the capabilities of `npm link`, but what I use it for is when I already have a dependency, say `uid`, and I want to hack on it inside my main app, I would `npm link` in that separate `uid` project folder, then inside my main app, I'd run `npm link uid`, so the changes I make in my main app, are also reflected in my `uid` project folder.

If that's the workflow you're describing with `npm link`, then `NODE_PATH` is different. `NODE_PATH` is useful when you have `uid` inside your main app. You've developed it inside because it's quick to iterate on and you're not sure it's going to make sense as something standalone. When you do decide to break it out from the project, you don't have to make any changes to the references in the current project.

**Comment by dominicbarnes** [comment]
Oh, you are talking about things **inside** the app's root directory still? Sorry, I was confused about exactly what we were talking about.

So, given a project structure like:

 - client/
    - app.js
 - lib/
    - uid/
       - index.js

You want `client/app.js` to do something like:

```js
var uid = require('uid');
```

So, you can use `lib` instead of something in `node_modules`? (but `uid` still lives inside your project)

**Comment by matthewmueller** [comment]
yep you got it :-)

**Comment by dominicbarnes** [comment]
Ok, I'll follow throw on exposing `options.resolveOptions` for both js and css modules tonight.

**Comment by dominicbarnes** [comment]
Done! (released with 0.6.1+)

**Comment by matthewmueller** [comment]
thanks man, really appreciate it :-)
